src/request.js

```diff
@@ -136,4 +136,26 @@ module.exports = class Request extends EventEmitter {
     }
     return null;
   }
+
+  // Temporarily suspends the flow of data from the database.
+  // No more 'row' events will be emitted until resume() is called.
+  pause() {
+    if (!this.isActive()) {
+      return;
+    }
+    this.connection.pauseTokenStreamParser();
+  }
+
+  // Resumes the flow of data from the database.
+  resume() {
+    if (!this.isActive()) {
+      return;
+    }
+    this.connection.resumeTokenStreamParser();
+  }
+
+  // Returns true if this request is the currently active request of the connection.
+  isActive() {
+    return this.connection && this.connection.request === this && this.connection.state === this.connection.STATE.SENT_CLIENT_REQUEST;
+  }
 };
```

Comment:
If we have multiple active result sets (MARS) in the future, we need pause/resume per request. But you are right, the deep coupling is not clean. I will move this logic into the Connection class, but keep the pause/resume methods in the Request class.

It's a general problem that we cannot indicate public/protected/private with class methods and properties. Therefore is suggest to use TypeScript (see #512).